modules/server/src/network/aggregations/AggregationAccumulator.ts

```diff
@@ -1,32 +1,62 @@
+import { ALL_NETWORK_AGGREGATION_TYPES_MAP } from '..';
 import { SUPPORTED_AGGREGATIONS } from '../common';
 import { Aggregations, Bucket, NumericAggregations } from '../types/aggregations';
+import { Hits } from '../types/hits';
 import { AllAggregations } from '../types/types';
+import { RequestedFieldsMap } from '../util';
 
 type ResolveAggregationInput = {
-	aggregationsMap: AllAggregations;
+	data: { aggregations: AllAggregations; hits: Hits };
 	accumulator: AllAggregations;
+	requestedFields: string[];
 };
 
 type AggregationsTuple = [Aggregations, Aggregations];
 type NumericAggregationsTuple = [NumericAggregations, NumericAggregations];
 
+const emptyAggregation = (hits: number) => ({
+	bucket_count: 1,
+	buckets: [{ key: '___aggregation_not_available___', doc_count: hits }],
+});
+
+// mutation - update a single aggregations field in the accumulator
+const addToAccumulator = ({ existingAggregation, aggregation, type }) => {
+	// if first aggregation, nothing to resolve with yet
+	return !existingAggregation
+		? aggregation
+		: resolveToNetworkAggregation(type, [aggregation, existingAggregation]);
+};
+
 /**
  * Resolves returned aggregations from network queries into single accumulated aggregation
  *
  * @param
  */
-const resolveAggregations = ({ aggregationsMap, accumulator }: ResolveAggregationInput) => {
-	Object.keys(aggregationsMap).forEach((fieldName) => {
-		const aggregation = aggregationsMap[fieldName];
-		const aggregationType = aggregation?.__typename || '';
-
-		const accumulatedFieldAggregation = accumulator[fieldName];
-
-		// mutation - update a single aggregations field in the accumulator
-		// if first aggregation, nothing to resolve yet
-		accumulator[fieldName] = !accumulatedFieldAggregation
-			? aggregation
-			: resolveToNetworkAggregation(aggregationType, [aggregation, accumulatedFieldAggregation]);
+const resolveAggregations = ({ data, accumulator, requestedFields }: ResolveAggregationInput) => {
+	requestedFields.forEach((requestedField) => {
```

Comment:
iterate over requested fields instead of node response data to ensure all properties requested have _some_ value
eg. bucket data or missing bucket data